pages/blog/api-first-with-asyncapi.md

```diff
@@ -0,0 +1,284 @@
+      name: turnOnOff
+      title: Turn on/off
+      summary: Command a particular streetlight to turn the lights on or off.
+      headers:
+        type: object
+        properties:
+          my-app-header:
+            type: string
+      payload:
+        $ref: "#/components/schemas/turnOnOffPayload"
+
+```
+
+### Message Payloads / Schemas
+
+You can define message payloads as:
+
+- Inline components/schemas in the same familiar way you do in OpenAPI
+- External files: both `json-schema` and `avro schemas` (.avsc) are supported
+
+```yml
+components:
+  messages:
+    MessageWithAsyncAPISchema:
+      payload:
+        $ref: "#/components/schemas/turnOnOffPayload" ## asyncapi/inline schema
+    MessageWithExternalJsonSchema:
+      schemaFormat: 'application/schema+json;version=draft-07'
+      payload:
+        $ref: "some/external/file.schema" ## a json-schema file
+    MessageWithAvroSchema:
+      schemaFormat: application/vnd.apache.avro+json;version=1.9.0
+      payload:
+        $ref: "v1/imports/file.avsc" ## and avro schema file
+```
+
+
+### Reusing Configurations: Operation Traits, Message Traits...
+
+Operation Traits, Message Traits are an excellent way to reuse chunks of configuration between different operations or messages.
+
+For instance if various messages share some common headers, you can configure them as Message Traits:
+```yml
+components:
+  messages:
+    CustomerEventMessage:
+      name: CustomerEventMessage
+      title: Async Event for a Customer
+      summary: Async Event for a Customer
+      schemaFormat: application/vnd.aai.asyncapi;version=2.4.0
+      traits:
+      - $ref: '#/components/messageTraits/CommonHeaders' # 'CommonHeaders' contents will replace 'traits' property
+      payload:
+      $ref: '#/components/schemas/CustomerEventPayload'
+
+  messageTraits:
+    CommonHeaders:
+      headers:
+      type: object
+      properties:
+      my-app-header:
+        type: integer
+        minimum: 0
+        maximum: 100
+```
+
+And the same concept applies to Operation Traits.
+
+
+## Different Styles of Event Messages
+
+### Notification Messages
+
+An Event Notification **contains minimal information about the event** and enough information for interested consumers to locate additional details. The specifics of what information is included in an event notification can vary depending on the system or use case.
+
+```json
+{
+  "headers": {
+    "event-type": "customer-created",
+    "event-id": "",
+    "aggregate-id": "1",
+    "aggregate-type": "customer"
+  },
+  "payload": {
+    "id": 1,
+    "eventType": "created",
+    "link": "/customers/1"
+  }
+}
+```
+
+### State Transfer Messages
+
+On the other hand a State Transfer message **contains the entire state of the aggregate** so consumer does not need to make additional calls. This can be useful in situations where subscribers need to maintain a synchronized view of the data. Compacted keyed topics typically use this style of messages.
```

Comment:
```suggestion
On the other hand, a State Transfer message **contains the entire state of the aggregate**, so a consumer does not need to make additional calls. This can be useful in situations where subscribers need to maintain a synchronized view of the data. Compacted keyed topics typically use this style of messages.
```